### DISCUSSION TITLE
Improve states output format

### DISCUSSION THREAD

**Discussion by origamimagiro** [origamimagiro/flat-folder#21] [Ideas]
Currently, we compute and store each assignment found for each component in an array. We should be able to considerably compress this output by storing a decision tree instead, only storing each guess list once, preferably branching on larger guess lists first. May also be able to use such information to identify other components that are disconnected. This discussion is for identifying and evaluating possible output structures. We would want to be able to store both branching to different incompatible assignments, as well as to components that can be assigned independently.

**Comment by OrigamiDraw** [comment]
I think the best visualization for most origamists would be an exploded view of the model in different states but that would be very difficult to implement. Short of that, I think cross-sections would be helpful. I made some for discussion #20 and would like to show them here too:
<img width="683" alt="Cross-sections" src="https://user-images.githubusercontent.com/110136377/182048892-63602ac9-1e4f-4e66-be98-da6fbf1bb542.png">
Using such cross-sections, most users should be able to specify additional constraints that would solve the CP.
